pkg/agent/plugin/workloadattestor/unix/unix_posix.go

```diff
@@ -176,13 +176,15 @@ func (p *Plugin) Attest(_ context.Context, req *workloadattestorv1.AttestRequest
 		selectorValues = append(selectorValues, makeSelectorValue("path", processPath))
 
 		if config.WorkloadSizeLimit >= 0 {
-			exePath := p.getNamespacedPath(proc)
-			sha256Digest, err := util.GetSHA256Digest(exePath, config.WorkloadSizeLimit)
-			if err != nil {
-				return nil, status.Error(codes.Internal, err.Error())
+			exePath, err := p.getNamespacedPath(proc)
+			if err == nil {
```

Comment:
error will be ignored, can you return it instead?

Reply:
Can do, but wouldn't it more useful to still return the list of selectors that have been attested about the process even if we got some errors along the way? 

We could log the errors so admins are aware of the issue, but return what we have since those selectors might be enough for the workload to function.

Reply:
>Can do, but wouldn't it more useful to still return the list of selectors that have been attested about the process even if we got some errors along the way?

Workload attestors in SPIRE take a conservative approach when there are selectors that can't be discovered, and if there is an unexpected condition that prevents one of the selectors from being discovered, then we usually fail. The important part to consider is if this error would be something expected under normal conditions.
In this case, the `discover_workload_path` configurable is explicitly telling to SPIRE if the user is expecting to be able to collect the `unix:path` and `unix:sha256` selectors or not.
We have this documented in the plugin documentation:
>If configured with discover_workload_path = true, the plugin will discover the workload path to provide additional selectors. If the plugin cannot discover the workload path or gather selectors based on the path, it will fail the attestation attempt.

I suggest to fail the attestation if there is an error here.